src/dodal/devices/zocalo/zocalo_results.py

```diff
@@ -0,0 +1,281 @@
+                        "dtype": "number",
+                        "shape": [],
+                    },
+                ),
+                (
+                    self._name + "-z_position",
+                    {
+                        "source": f"sim://{self._prefix}",
+                        "dtype": "number",
+                        "shape": [],
+                    },
+                ),
+                (
+                    self._name + "-bbox_size_x",
+                    {
+                        "source": f"sim://{self._prefix}",
+                        "dtype": "number",
+                        "shape": [],
+                    },
+                ),
+                (
+                    self._name + "-bbox_size_y",
+                    {
+                        "source": f"sim://{self._prefix}",
+                        "dtype": "number",
+                        "shape": [],
+                    },
+                ),
+                (
+                    self._name + "-bbox_size_z",
+                    {
+                        "source": f"sim://{self._prefix}",
+                        "dtype": "number",
+                        "shape": [],
+                    },
+                ),
+                (
+                    self._name + "-results_valid",
+                    {
+                        "source": f"sim://{self._prefix}",
+                        "dtype": "boolean",
+                        "shape": [],
+                    },
+                ),
+            ],
+        )
+
+    def _get_zocalo_connection(self):
+        zc = zocalo.configuration.from_file()
+        zc.activate_environment(self.zocalo_environment)
+
+        transport = lookup("PikaTransport")()
+        transport.connect()
+        return transport
+
+    def _subscribe_to_results(self):
+        transport = self._get_zocalo_connection()
+
+        def _receive_result(
+            rw: workflows.recipe.RecipeWrapper, header: dict, message: dict
+        ) -> None:
+            LOGGER.info(f"Received {message}")
+            recipe_parameters = rw.recipe_step["parameters"]  # type: ignore # this rw is initialised with a message so recipe step is not None
+            LOGGER.info(f"Recipe step parameters: {recipe_parameters}")
+            transport.ack(header)
+
+            results = message.get("results", [])
+            call_in_bluesky_event_loop(
+                self._raw_results_received.put(results)
+            )  # TODO is this the right way to do this? this means these devices can't be used outside plans
+
+        workflows.recipe.wrap_subscribe(
+            transport,
+            self.channel,
+            _receive_result,
+            acknowledgement=True,
+            allow_non_recipe_messages=False,
+        )
+
+
+def trigger_zocalo(zocalo: ZocaloResults):
+    """A minimal utility plan which will wait for analysis results to be returned from
+    Zocalo, and bundle them in a reading."""
```

Comment:
Should: I'm not sure if trigger is the right word here, it kind of implies it's just starting it and not waiting for the end. I think ultimately in Hyperion we might want to separate start/end more to do parallel processing but we can think about that when we get to it.